op-node/rollup/derive/frame.go

```diff
@@ -130,6 +133,29 @@ func ParseFrames(data []byte) ([]Frame, error) {
 	if len(data) == 0 {
 		return nil, errors.New("data array must not be empty")
 	}
+	// If the calldata starts with the function signature of the submit method
+	// then it is a v2 frame.
+	// TODO: don't parse the ABI every time
```

Comment:
There's an example of caching the ABI in the batcher, on the op_batcher specific Config.

Reply:
I went with a module level global for the ABI. I think the ABI should always be available and if it isn't the program exiting early is not a bad idea.